README.md

```diff
@@ -672,13 +672,28 @@ d3.reverse(new Set([0, 2, 3, 1])) // [1, 3, 2, 0]
 ```
 
 <a name="sort" href="#sort">#</a> d3.<b>sort</b>(<i>iterable</i>, <i>comparator</i> = d3.ascending) · [Source](https://github.com/d3/d3-array/blob/master/src/sort.js)
+<br><a name="sort" href="#sort">#</a> d3.<b>sort</b>(<i>iterable</i>, <i>accessor</i>)
 
-Returns an array containing the values in the given *iterable* in the sorted order defined by the given *comparator* function. If *comparator* is not specified, it defaults to [d3.ascending](#ascending). Equivalent to [*array*.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort), except that it does not mutate the given *iterable*, and the comparator defaults to natural order instead of lexicographic order:
+Returns an array containing the values in the given *iterable* in the sorted order defined by the given *comparator* or *accessor* function. If *comparator* is not specified, it defaults to [d3.ascending](#ascending). Equivalent to [*array*.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort), except that it does not mutate the given *iterable*, and the comparator defaults to natural order instead of lexicographic order:
 
 ```js
 d3.sort(new Set([0, 2, 3, 1])) // [0, 1, 2, 3]
 ```
 
+If an *accessor* (a function that takes a single argument) is specified,
+
+```js
+d3.sort(data, d => d.value)
+```
+
+it is equivalent to a *comparator* using [natural order](#ascending):
+
+```js
+d3.sort(data, (a, b) => d3.ascending(a.value, b.value))
+```
+
+The *accessor* is only invoked once per element, and thus may be nondeterministic.
```

Comment:
With this we could almost use the initial index: `d3.sort(A, function(d) { return -arguments[1]; })` works—the only caveat is we can't write `function(d,i)` since it would have a length===2.

Reply:
Sorry, but I’m not understanding this comment. This:

```js
d3.sort(A, function(d) { return -arguments[1]; })
```

This is an elaborate equivalent to this?

```js
d3.reverse(A)
```

To avoid the length ambiguity we could introduce a separate method, say d3.sortBy, that takes an accessor rather than a comparator. But I think it’s convenient to use the same length test that d3.bisector used. I don’t think you’ll often want to reference the original index in an accessor, and if you did, you could always just do the sorting yourself.

Reply:
Yes the example was too simple—what I had in mind was more "sorting with reference to an external array of values". But it was more a comment in passing than a specific concern.